libraries/stdlib/common/src/generated/_Arrays.kt

```diff
@@ -5295,13 +5285,16 @@ public inline fun CharArray.takeLastWhile(predicate: (Char) -> Boolean): List<Ch
  * @sample samples.collections.Collections.Transformations.take
  */
 public inline fun <T> Array<out T>.takeWhile(predicate: (T) -> Boolean): List<T> {
```

Comment:
it will be easier to read, i think, with equivalent performance 
```suggestion
public inline fun <T> Array<out T>.takeWhile(predicate: (T) -> Boolean): List<T> {
    var i = 0
    while (i < size && predicate(this[i])) i++
    return if (i == 0) emptyList() else this.copyOfRange(0, i).asList()
}
```

Reply:
I've checked the impact of removing the special case handling, and it looks like it is noticeable. On the benchmark result below, your suggestion is the "takeWhileSuggestion" case. We see that for array size 0 and 1, it is a little less fast than the proposed optimization : 

![image](https://github.com/user-attachments/assets/6658ed62-7e86-405e-882d-14aeb94e6e7b)

Therefore, I think we should keep the when as it is now.

Reply:
Nice @alexismanin

Reply:
I've pushed the change to `takeWhile` function.